sys/riscv/riscv/trap.c

```diff
@@ -582,13 +591,6 @@ do_trap_user(struct trapframe *frame)
 		    (uintcap_t)frame->tf_stval, exception, 0);
 		userret(td, frame);
 		break;
-	case SCAUSE_STORE_AMO_CAP_PAGE_FAULT:
-		if (log_user_cheri_exceptions)
-			dump_cheri_exception(frame);
-		call_trapsignal(td, SIGSEGV, SEGV_STORETAG,
```

Comment:
The `SIGSEGV` and `SEGV_STORETAG` functionality appears lost.  vm_fault() would need to be changed to return these values if it fails with `ftype` containing `VM_PROT_WRITE_CAP`.